### PR TITLE
Refactoring/jr2 1294 candidate dtos

### PR DESCRIPTION
**Pull request by cjpsycos**

### CODE REVIEW

File: candidate-app/src/main/java/ch/admin/seco/service/candidate/service/dto/CandidateProtectedDataCreateDto.java [hunk 1]
```diff
@@ -0,0 +1,130 @@
...
+import java.time.LocalDate;
+import java.util.UUID;
+
+import javax.validation.Valid;
+import javax.validation.constraints.NotNull;
+import javax.validation.constraints.Size;
+
+public class CandidateProtectedDataCreateDto {
+
+    private UUID id;
```
Comment (cjpsycos) [inline candidate-app/src/main/java/ch/admin/seco/service/candidate/service/dto/CandidateProtectedDataCreateDto.java]: why does the create DTO has an id? The id should be set on persisting on the db or on creation of the domain object.

File: candidate-app/src/main/java/ch/admin/seco/service/candidate/service/impl/CandidateServiceImpl.java [hunk 2]
```diff
@@ -105,11 +100,16 @@ public CandidateProtectedDataDto save(CandidateProtectedData candidate) {
         candidate.setCandidateProfile(null);
```
Comment (cjpsycos) [inline candidate-app/src/main/java/ch/admin/seco/service/candidate/service/impl/CandidateServiceImpl.java]: is save used only for creation, then rename it to createCandidateProtectedData(dto DTO). on the service layer it should receive a dto and give back a dto/id

File: candidate-app/src/main/java/ch/admin/seco/service/candidate/service/impl/CandidateServiceImpl.java [hunk 3]
```diff
@@ -105,11 +100,16 @@ public CandidateProtectedDataDto save(CandidateProtectedData candidate) {
...
         candidate = candidateProtectedDataRepository.save(candidate);
         if (Objects.nonNull(profile)) {
-            save(profile.id(candidate.getId()));
+            saveCandidateProfile(profile.id(candidate.getId()));
```
Comment (cjpsycos) [inline candidate-app/src/main/java/ch/admin/seco/service/candidate/service/impl/CandidateServiceImpl.java]: same here

File: candidate-app/src/main/java/ch/admin/seco/service/candidate/web/rest/CandidateResource.java [hunk 4]
```diff
@@ -75,45 +77,28 @@ public CandidateResource(CandidateService candidateService) {
...
-    public ResponseEntity<CandidateProtectedDataDto> createCandidate(@Valid @RequestBody CandidateProtectedData candidate) throws URISyntaxException {
+    public ResponseEntity<CandidateProtectedDataDto> createCandidate(@Valid @RequestBody CandidateProtectedDataCreateDto candidate) throws URISyntaxException {
         log.debug("REST request to save Candidate : {}", candidate);
         if (candidate.getId() != null) {
             return ResponseEntity.badRequest()
                 .headers(HeaderUtil.createFailureAlert(ENTITY_NAME, "idexists", "A new candidate cannot already have an ID"))
                 .body(null);
         }
-        CandidateProtectedDataDto result = candidateService.save(candidate);
+        CandidateProtectedDataDto result = candidateService.save(CandidateProtectedDataMapper.convertToEntity(candidate));
```
Comment (cjpsycos) [inline candidate-app/src/main/java/ch/admin/seco/service/candidate/web/rest/CandidateResource.java]: conversion should be done in the service

File: candidate-app/src/main/java/ch/admin/seco/service/candidate/web/rest/CandidateResource.java [hunk 5]
```diff
@@ -75,45 +77,28 @@ public CandidateResource(CandidateService candidateService) {
...
             .body(result);
     }
 
     @PatchMapping("/candidates")
     @Timed
-    public ResponseEntity<Void> updateCandidates(@Valid @RequestBody Collection<CandidateProtectedData> candidates) {
+    public ResponseEntity<Void> updateCandidates(@Valid @RequestBody Collection<CandidateProtectedDataCreateDto> candidates) {
         log.debug("REST request to save Candidates : {}", candidates);
-        candidateService.save(candidates);
+        candidateService.save(candidates.stream().map(CandidateProtectedDataMapper::convertToEntity)
```
Comment (cjpsycos) [inline candidate-app/src/main/java/ch/admin/seco/service/candidate/web/rest/CandidateResource.java]: same here

File: candidate-app/src/main/java/ch/admin/seco/service/candidate/service/dto/CandidateProtectedDataCreateDto.java (hunk 1 above)
Comment (pavax) [reply candidate-app/src/main/java/ch/admin/seco/service/candidate/service/dto/CandidateProtectedDataCreateDto.java]: I guess the create is also used for update and thus it needs an id.... but I'm not sure

**Pull request closed without merging**